src/iter.rs

```diff
@@ -277,6 +277,28 @@ impl<W: tokio::io::AsyncBufRead + Unpin + tokio::io::AsyncRead> tokio::io::Async
     }
 }
 
+#[cfg(feature = "futures")]
+#[cfg_attr(docsrs, doc(cfg(feature = "futures")))]
+impl<S: futures::Stream + Unpin> futures::Stream for ProgressBarIter<S> {
+    type Item = S::Item;
+
+    fn poll_next(
+        self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        let this = self.get_mut();
+        let item = std::pin::Pin::new(&mut this.it).poll_next(cx);
+
+        if item.is_ready() {
```

Comment:
This seems wrong? As documented `poll_next()` will yield `Ready` many times, with the `Ready` including `Some` while still going and then yielding `Ready(None)` when the stream has been exhausted. So I think this should be:

```rust
match &item {
    Poll::Ready(Some(_)) => this.progress.inc(1),
    Poll::Ready(None) => this.progress.finish_using_style(),
    Poll::Pending => {}
}
```

Reply:
Haha, that's a bit embarassing - I'll make that change